File: python/010.py
#!/usr/bin/env python3

from math import sqrt

def main():
  """ this computes projecteuler.net problem 010 """

  last = 2000000;
  primes = [2]

  psum = 2
  for test in range(3, last, 2):
    if (is_prime(test, primes)):
      primes.append(test);
      psum += test

  print(psum)

def is_prime(test, primes):
  root = sqrt(test);

  for i in primes:
    if (i > root):
      return 1;
    if ((test % i) == 0):
      return 0;

if __name__ == "__main__":
    main()

File: perl/010.pl
#!/usr/bin/perl -w
use strict;

my $last = 2000000;

my @primes = (2);
my $sum = 2;
for (my $test=3; $test < $last; $test += 2) {
  if (is_prime($test, \@primes)) {
    push @primes, $test;
    $sum += $test;
  }
}

print $sum,"\n";

sub is_prime {
  my $test = shift;
  my $known_primes_ref = shift;

  my $root = sqrt($test);
  foreach my $number (@$known_primes_ref) {
    return 1 if ($number > $root);
    return 0 unless ($test % $number);
  }
  return 1;
}
